my_package/activelearning.py

```diff
@@ -0,0 +1,63 @@
+"""
+Runnable script for the active learning pipeline
+"""
+import sys
+from dataset.dataset import BraninDatasetHandler, Branin_Data
+from surrogate.surrogate import SingleTaskGPRegressor
+from sampler.sampler import Sampler, RandomSampler
+from filter.filter import Filter, OracleFilter
+from gflownet.proxy.box.branin import Branin
+
+import torch
+device = "cuda" if torch.cuda.is_available() else "cpu"
+import numpy as np
+
+n_iterations = 5 # TODO: replace with budget
+grid_size = 10
+float_prec = 64
+n_samples = 5
+train_path = "./storage/branin/data_%i_train.csv"%grid_size
+
+def main():
+    # define candidate set
+    xi = np.arange(0,grid_size)
+    yi = np.arange(0,grid_size)
+    grid = np.array(np.meshgrid(xi,yi))
+    grid_flat = torch.tensor(grid.T, dtype=torch.float64).reshape(-1,2).to(device)
+    candidate_set, _ = Branin_Data(grid_flat, grid_size=grid_size, device=device)[:]
+    
+    
+    # Dataset
+    dataset_handler = BraninDatasetHandler(train_path=train_path, train_fraction=1.0, device=device, float_precision=float_prec)
+    # Oracle
+    oracle = Branin(fidelity=1, do_domain_map=True, device=device, float_precision=float_prec)
+    # Filter
+    filter = OracleFilter(oracle)
+
+
+    for i in range(n_iterations):
```

Comment:
I would suggest to make the active learning loop depend on a "budget", rather than a number of iterations. The reason is that when we consider multi-fidelity active learning, each oracle will have a cost and not all iterations will incur the same cost. So something like:

```
while budget_used < budget:
    # Do active learning stuff
```